src/api/java/com/minecolonies/api/crafting/IItemStorageFactory.java

```diff
@@ -0,0 +1,50 @@
+package com.minecolonies.api.crafting;
+
+import com.minecolonies.api.colony.requestsystem.factory.FactoryVoidInput;
+import com.minecolonies.api.colony.requestsystem.factory.IFactory;
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import static com.minecolonies.api.util.constant.Constants.PARAMS_ITEMSTORAGE;
+
+/**
+ * Interface for the IItemStorageFactory which is responsible for creating and maintaining ItemStorage objects.
+ */
+public interface IItemStorageFactory extends IFactory<FactoryVoidInput, ItemStorage>
```

Comment:
This is for an entire different usecase, it's so that we can serialize itemStorages better.
Nothing related directly to research.
We still will need this, always.

Reply:
okay not research related. i thought it was